extension/jaegerremotesampling/internal/source/filesource/filesource.go

```diff
@@ -156,6 +160,9 @@ func (h *samplingProvider) samplingStrategyLoader(strategiesFile string) strateg
 }
 
 func (h *samplingProvider) autoUpdateStrategies(ctx context.Context, interval time.Duration, loader strategyLoader) {
+	h.wg.Add(1)
```

Comment:
Add() should be called outside of the goroutine itself, otherwise you can get to a race condition where Close and Wait succeed before goroutine starts.